### ISSUE TITLE
Update validation related to overlap and projections

### ISSUE THREAD

**Issue by emlys** [natcap/invest#624]
This came up from a conversation on slack about how to include `projected`, `projection_units`, and `args_with_spatial_overlap` info in the generated docs. Here's just some thoughts, I'm not totally sold on this and happy to discuss.
### Remove the `args_with_spatial_overlap` key
It seems safe to assume that all of a model's spatial inputs (at least non-nested inputs) should overlap. So in validation we can check overlap for all the args that are `raster`s or `vector`s. @davemfish  points out that
  > there are cases it could conceivably be okay to have inputs with small bounding boxes that don't intersect each other (e.g. CV habitats, Rec predictors, HRA layers). That said, all those spatial inputs are probably nested within another arg anyway and not directly in the spec. ... There are validation functions in Rec & CV for those inputs, I believe. Not sure if they check overlap, or just check for projections, units, whatever the model needs.

### Remove the `projection_units` key
We use `projection_units` to enforce that the raster's pixels are measured in meters, usually so we can convert to hectares. Now that we're using `pint`, this is kind of an unnecessary restriction because we can get the raster's units from the metadata and convert them to meters or hectares very easily. Any model that relies on meters could get an update to convert any linear unit.

### Replace the `different_projections_ok` and `projected` keys with a new `projections` section
A lot of models need a projected coordinate system because they do distance or area operations that assume each pixel is the same size. About half of our models use `pygeoprocessing.align_and_resize_raster_stack` to reproject all spatial inputs to the projection of one "important" one like the LULC or DEM. This seems like the preferred design for new models and it's nice to require `"projected": True` for as few inputs as possible. It would also be nice to document which input this target projection comes from. Some model UG pages mention it but that's something we could generate as well.

#### for models that don't reproject
Assume that either all inputs need to be projected, or none need to be. Move the `projected` key up to the model level of the args spec and default it to `True`.
Example:
```
ARGS_SPEC = {
    ...
    "projections": {
        "reprojects": False,
        "all_must_be_projected": [True/False]}
```
#### for models that do reproject
Assume that one input has the "target" projection which all other inputs will be reprojected to, and all outputs will have.
```
ARGS_SPEC = {
    ...
    "projections": {
        "reprojects": True,
        "target_projection_key": "dem_path",
        "target_must_be_projected": [True/False]}
```

**Comment by davemfish** [comment]
I think it would be good to narrow down on the problem with respect to generating documentation, since I think that is what inspired this issue. Is there anything wrong with using the data as-is? I guess the main thing missing is being able to generate a note about which dataset is used as a target projection when the model re-projects? 

From the perspective of validation, these parts of the `ARGS_SPEC` and `validation` have been around since `3.8.0` and I think they're working well.

>Any model that relies on meters could get an update to convert any linear unit.

I think this is probably true, but I'd say we should only do this if it seems important for us to loosen the restriction on input projections. And I think that could be a separate issue based on user-demand.

**Comment by emlys** [comment]
You're right this is kinda vague. I think one problem is that the values of `spatial_keys` and which args are `projected` are incomplete or wrong in some models and don't line up with what the user's guide says. As far as generating documentation, I think we should be able to document which input's projection we reproject to, and which projection the outputs inherit.
